### PR TITLE
Update website development doc

### PR DESCRIPTION
**Pull request by shifucun**
- Re-structure the docs by environment: local, dev GCP instance. This makes it straightforward to start the local development without jumping around.
- Remove local Minikube development instruction: it's very lengthy process, easier to fail due to local resource constraints and thus bare used. Deploying to dev instance could serve the same purpose.

### CODE REVIEW

File: docs/developer_guide.md [hunk 1]
```diff
@@ -46,25 +49,67 @@ requirements:
...
+The following example will start localhost on port 8081. The default is 8080.
+
+```bash
+./run_server.sh -p 8081
+```
+
+Please note the strict syntax requirements for the script, and leave a space
+after the flag. So: `./run_server.sh -p 8081` but not `./run_server.sh -p=8081`.
+
+## Deploy local instance to GCP (dev insance)
```
Comment (chejennifer) [inline docs/developer_guide.md:90]: dev instance

File: docs/developer_guide.md [hunk 2]
```diff
@@ -46,25 +49,67 @@ requirements:
...
+Then run
+
+```bash
+gcloud auth login
+gcloud auth configure-docker
+./scripts/push_image.sh
+./scripts/deploy_gke.sh dev us-central1
+```
+
+The script builds docker image locally and tag it with the local git commit
```
Comment (chejennifer) [inline docs/developer_guide.md:102]: nit: ...and tags it with ..., then deploys to dev ...

**Review by chejennifer — APPROVED** [review]
thanks for updating the docs!

File: docs/developer_guide.md (hunk 1 above)
Comment (shifucun) [reply docs/developer_guide.md]: Done

File: docs/developer_guide.md [hunk 3]
```diff
@@ -9,29 +9,32 @@ containers:
...
+## Local Development with Flask
 
-### Basic Development
+For changes that do not test GCP deployment or involve mixer changes, can
+simply run in local environment (Mac or Linux machine). This way the local Flask
+App talks to the [autopush Mixer](https://autopush.api.datacommons.org).
 
-For development that only involves minor bug fixes or feature additions, the
-following steps are required:
+Note: the `autopush mixer` contains the latest data and mixer code change. It
```
[the remark this answers is not in the document]
Comment (shifucun) [reply docs/developer_guide.md]: Done

File: docs/developer_guide.md (hunk 2 above)
Comment (shifucun) [reply docs/developer_guide.md]: Done

**Pull request merged**